### DISCUSSION TITLE
gg: 像写 Golang 一样生成代码

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#252] [Announcements]
# gg: 像写 Golang 一样生成代码

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/2021/09-gg/

**Comment by lilydjwg** [comment]
看起来是，字符串模板适合以字符串为主的情况（比如 HTML），而构造器适合逻辑多变、字符串很少的情况（比如几乎啥都要通过外部数据生成的代码）。

**Comment by STRRL** [comment]
chaos-mesh 里也大量使用了代码生成，而且还曾经在 bugbash 的活动中，大量的使用 `text/template` 被机器人认为是 bad smell, 要换成 `html/template` 🤣

**Comment by disksing** [comment]
漩涡gg

**Comment by bluntdel** [comment]
看起来gg的可读性也堪忧啊，而且还有本身的学习成本。  本来看一眼模板就知道大概咋生成啥代码，这里还要演变成读一段逻辑代码了。  这就变成在写代码生成的代码了，感觉略有点本末倒置了，这得有多大的重复代码量才需要这么干。。。而且修改成本也挺高

**Reply by Xuanwo** [reply]
嗯，就像前面依云说的一样，模板适合字符串为主的场景，gg 这样的构造器适合逻辑复杂的场景，有大量的数据要传入。中间的 tradeoff 还是需要好好斟酌的，大多数时候模板已经足够啦～
